shell/platform/fuchsia/flutter/pointer_injector_delegate.h

```diff
@@ -106,7 +106,11 @@ class PointerInjectorDelegate {
             if (!weak) {
               return;
             }
-            weak->registered_ = false;
+
+            // Clear all the stale pointer events in |injector_events_| and
+            // reset the state of |weak| so that any future calls do not inject
+            // any stale pointer events.
+            weak->Reset();
```

Comment:
is there a way for a test to observe that the queue flush happened, after a Reset()?

Reply:
Its difficult to control how the pointer events are buffered. Right now the only way to dispatch a pointer event is through HandlePlatformMessage. HandlePlatformMessage flushes the queue and dispatches the event immediately. So even when we close the device channel through the test, it is not guaranteed that there would be some pointer events in the buffer at that time.